src/UiTPAS/Validation/EventHasTicketSalesGuard.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\UiTPAS\Validation;
+
+use Broadway\Repository\AggregateNotFoundException;
+use CultureFeed_Uitpas;
+use CultuurNet\UDB3\Event\Event;
+use CultuurNet\UDB3\Event\EventRepository;
+use CultuurNet\UDB3\Offer\Commands\AbstractCommand;
+use CultuurNet\UDB3\Offer\Commands\DeleteOrganizer;
+use CultuurNet\UDB3\Offer\Commands\UpdateOrganizer;
+use Psr\Log\LoggerInterface;
+
+final class EventHasTicketSalesGuard
+{
+    private CultureFeed_Uitpas $uitpas;
+
+    private EventRepository $eventRepository;
+
+    private LoggerInterface $logger;
+
+    public function __construct(
+        CultureFeed_Uitpas $uitpas,
+        EventRepository $eventRepository,
+        LoggerInterface $logger
+    ) {
+        $this->uitpas = $uitpas;
+        $this->eventRepository = $eventRepository;
+        $this->logger = $logger;
+    }
+
+    public function guard(AbstractCommand $command): void
+    {
+        if (!($command instanceof UpdateOrganizer) &&
+            !($command instanceof DeleteOrganizer)) {
+            return;
+        }
+
+        $eventId = $command->getItemId();
+
+        try {
+            /** @var Event $event */
+            $event = $this->eventRepository->load($eventId);
+        } catch (AggregateNotFoundException $exception) {
+            // The command is for a place which can not have ticket sales.
+            return;
+        }
+
+        $equalOrganizers = $event->getOrganizerId() === $command->getOrganizerId();
+        if ($command instanceof UpdateOrganizer && $equalOrganizers) {
+            // A change only happens when a different organizer is updated
+            return;
+        }
+        if ($command instanceof DeleteOrganizer && !$equalOrganizers) {
+            // A change only happens when the existing organizer is deleted
+            return;
+        }
+
+        try {
+            $hasTicketSales = $this->uitpas->eventHasTicketSales($eventId);
+        } catch (\Exception $exception) {
```

Comment:
I think if we catch `Throwable` here instead of `Exception`, we could fix https://jira.uitdatabank.be/browse/III-4721 at the same time.

(Sentry calls it an `ErrorException` but I think it's actually just an `Error`, which is not an `Exception`)